app/ui/browser-modern/sagas/page-sagas.js

```diff
@@ -77,6 +79,9 @@ export function* destroyPageSession({ id, options = {} }) {
 
   if (!options.withoutUI) {
     const currentPageCount = yield select(PageSelectors.getPageCount);
+    const sessionId = yield apply(userAgentHttpClient, userAgentHttpClient.waitForSession, [page]);
+
+    yield put(PageActions.saveRestorablePage(page.id, sessionId));
```

Comment:
If you're feeling particularly brave, changing `page.id` to `page.uuid` would go a long way to improving clarity.